### PR TITLE
Feat/debug info screen

### PR DESCRIPTION
**Pull request by MasterKenth**
## Explain the changes you’ve made

Added a debug info screen.

## Explain why these changes are made

Useful for debugging.

## Explain your solution

Added a new screen similar in structure to the dev features screen. It can be accessed from the profile screen. It gathers information from different sources (currently system info and storage info but is extensible). 

## How to test

Concrete example:

1. Checkout this branch
2. Launch
3. Go to profile screen and click "Felsökningsinformation"
4. Verify things work

## Tested environments

- [] Storybook on a iOS device/simulator.
- [x] Building the Application on a iOS device/simulator.
- [x] Building the Application on a Android device/simulator.

## Screenshots

image

---

## Checklist for reviewers

Here are a few points that are useful for reviewers to check.

The short list of the (maybe) most important points are:

- Does this PR work as expected?
- Does the PR properly only concern a single feature/bug/thing?
- Is the code clear and concise?
- Will the changes impact existing functionality?
- Are there any linting errors (ESLint) that should be resolved?
- Do the automatic GitHub checks pass?

However it is good if a reviewer can be as thorough as possible,
within reasonable limits of course.

### Functionality

- Does this PR work as expected?
- If this is a new feature, does it add value, or is it a sign of feature-creep?
- Will the changes impact existing functionality?
- Does the changes create inconsistencies somewhere else?
- Does the code work on different platforms if applicable?
- How does the code handle if unexpected values are passed to functions etc.?

### Readability, syntax, formatting, style

- Is the code clear and concise?
- Has styling rules (Prettier) been applied to included files?
- Are functions/variables/etc. aptly named?

### Design

- Is the code properly planned and designed?
- Is code organized/separated in a logical fashion?
- Does the code follow clean code practices to a reasonable degree?
- Is the code self-documenting enough?
- Is the code in sync with existing patterns/technologies?
- Does the code reuse existing functionality where applicable?
- Is the code designed with reusability in mind?
- Does the code have hard-coded values that can be refactored?

### Patterns, idioms, tests, best practices

- Are there any linting errors (ESLint) that should be resolved?
- Is the code properly typed (TypeScript)?
- Does the code utilize language features to their full potential?

### Debugging, testing, configuring

- Is appropriate logging used?
- Have tests been added/modified to accommodate the changes?
- Is the feature sufficiently configurable in case tweaks are required in the future?

### Performance, reliability, scalability

- Is the code well optimized?
- Does the code scale?
- Does the code handle failure scenarios?

### Security

- Is there any obvious potential for exploits and security threats?
- Does code check proper authorization where applicable?

### Etiquettes

- Are commits atomic (does only one thing per commit)?
- Does the PR properly only concern a single feature/bug/thing?
- Are commit messages well-written?
- Do the automatic GitHub checks pass?

### CODE REVIEW

**Comment by helsingborgsstad** [comment]
Task linked: [CU-3cy7q1e Lägg till vy med felsökningsinformation](https://app.clickup.com/t/3cy7q1e)

File: source/hooks/useAsync.ts [hunk 1]
```diff
@@ -0,0 +1,26 @@
+import { useState, useEffect } from "react";
+import { to } from "../helpers/Misc";
+
+export default function useAsync<T>(
+  asyncFunction: () => Promise<T>
+): [boolean, T | undefined, Error | undefined] {
+  const [isLoading, setIsLoading] = useState(true);
```
Comment (kajm) [inline source/hooks/useAsync.ts:7]: Perhaps the asyncFunction should be a state instead. By doing this we avoid that the asyncFunction is re-triggered when the function is changed outside this hook.

File: source/screens/debugInfoScreen/DebugInfoScreen.styled.ts [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+`;
+
+export const MarginedText = styled.Text`
+  margin-top: 8px;
+  margin-bottom: 8px;
+`;
+
+export const Container = styled.ScrollView`
+  flex: 1;
+  padding-left: 16px;
```
Comment (kajm) [inline source/screens/debugInfoScreen/DebugInfoScreen.styled.ts:17]:
```js
margin: 16px 16px 40px 16px;
```

File: source/screens/debugInfoScreen/DebugInfoScreen.tsx [hunk 3]
```diff
@@ -0,0 +1,114 @@
+import React from "react";
+
```
Comment (kajm) [inline source/screens/debugInfoScreen/DebugInfoScreen.tsx:2]: reorder imports?

File: source/screens/debugInfoScreen/DebugInfoScreen.tsx [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+  CategoryEntryValue,
+  MarginButton,
+} from "./DebugInfoScreen.styled";
+import getDebugInfo from "../../helpers/debugInfo/debugInfo";
+
+interface Props {
+  navigation: {
+    navigate: (to: string) => void;
+  };
+}
```
Comment (kajm) [inline source/screens/debugInfoScreen/DebugInfoScreen.tsx:28]: no type file for this one?

File: source/screens/debugInfoScreen/DebugInfoScreen.tsx [hunk 5]
```diff
@@ -0,0 +1,114 @@
...
+        {error && (
+          <>
+            <MarginedText>
+              Ett fel uppstod vid insamling av felsökningsinformationen:
+            </MarginedText>
+            <MarginedText>{error.toString()}</MarginedText>
+          </>
+        )}
+
+        {debugInfo && (
```
Comment (kajm) [inline source/screens/debugInfoScreen/DebugInfoScreen.tsx:68]: **OPTIONAL**

Create a new component called DebugInfo and make that component decide to render content or not. By doing this we can avoid checks like:

```js
{debugInfo && ... }
```

**Review by kajm — CHANGES_REQUESTED** [review]

File: source/components/organisms/DebugInfo/DebugInfo.tsx [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+    Clipboard.setString(asString);
+
+    showNotification(
+      "Felsökning",
+      "Informationen har kopierats till urklippet.",
+      "success",
+      4000
+    );
+  };
+
```
Comment (kajm) [inline source/components/organisms/DebugInfo/DebugInfo.tsx:34]: if (!debugInfo) return null;

**Review by kajm — CHANGES_REQUESTED** [review]

File: source/components/organisms/DebugInfo/DebugInfo.styled.ts [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+import { Button } from "../../atoms";
+
+import type { ThemeType } from "../../../theme/themeHelpers";
+
+export const Container = styled.ScrollView`
+  flex: 1;
+`;
+
+export const InfoContainer = styled.View`
+  margin-top: 16px;
```
Comment (kajm) [inline source/components/organisms/DebugInfo/DebugInfo.styled.ts:12]:
```js
margin: 16px 24px 0px 24px;
```

File: source/components/organisms/DebugInfo/DebugInfo.styled.ts [hunk 8]
```diff
@@ -0,0 +1,49 @@
...
+  text-align: center;
+`;
+
+export const CategoryErrorText = styled.Text<{ theme: ThemeType }>`
+  color: ${({ theme }) => theme.colors.primary.red[1]};
+`;
+
+export const CategoryEntry = styled.View`
+  flex-direction: row;
+  margin-top: 2px;
```
Comment (kajm) [inline source/components/organisms/DebugInfo/DebugInfo.styled.ts:32]:
```js
margin: 2px 0px;
```

File: source/components/organisms/DebugInfo/DebugInfo.styled.ts [hunk 9]
```diff
@@ -0,0 +1,49 @@
...
+  text-align: right;
+  padding-right: 8px;
+`;
+
+export const CategoryEntryValue = styled.Text`
+  flex-basis: 60%;
+`;
+
+export const MarginButton = styled(Button)`
+  margin-top: 8px;
```
Comment (kajm) [inline source/components/organisms/DebugInfo/DebugInfo.styled.ts:47]:
```js
margin: 8px 0px;
```

File: source/components/organisms/DebugInfo/DebugInfo.test.tsx [hunk 10]
```diff
@@ -0,0 +1,13 @@
...
+
+describe("DebugInfo", () => {
+  it("renders", () => {
+    const func = () => {
+      render(<DebugInfo />);
+    };
+
+    expect(func).not.toThrow();
+  });
+});
```
Comment (kajm) [inline source/components/organisms/DebugInfo/DebugInfo.test.tsx:13]: Don't we want a test when we actually render something?

File: source/components/organisms/DebugInfo/DebugInfo.tsx [hunk 11]
```diff
@@ -0,0 +1,65 @@
+import Clipboard from "@react-native-clipboard/clipboard";
+import React from "react";
+
+import { useNotification } from "../../../store/NotificationContext";
+import { Text } from "../../atoms";
```
Comment (kajm) [inline source/components/organisms/DebugInfo/DebugInfo.tsx:5]: add newline

File: source/components/organisms/DebugInfo/DebugInfo.tsx [hunk 12]
```diff
@@ -0,0 +1,65 @@
...
+            {category.errorMessage && (
+              <CategoryErrorText>{category.errorMessage}</CategoryErrorText>
+            )}
+            {category.entries &&
+              category.entries.map((entry) => (
+                <CategoryEntry key={`${category.name}-${entry.name}`}>
+                  <CategoryEntryName>{entry.name}</CategoryEntryName>
+                  <CategoryEntryValue>{entry.value}</CategoryEntryValue>
+                </CategoryEntry>
+              ))}
```
Comment (kajm) [inline source/components/organisms/DebugInfo/DebugInfo.tsx:50]: **OPTIONAL**
Removing `category.entries &&` 

```js
              (category?.entries ?? []).map((entry) => (
                <CategoryEntry key={`${category.name}-${entry.name}`}>
                  <CategoryEntryName>{entry.name}</CategoryEntryName>
                  <CategoryEntryValue>{entry.value}</CategoryEntryValue>
                </CategoryEntry>
              ))
```

File: source/screens/debugInfoScreen/DebugInfoScreen.styled.ts [hunk 13]
```diff
@@ -0,0 +1,22 @@
...
+import { Button } from "../../components/atoms";
+import { ScreenWrapper } from "../../components/molecules";
+
+export const DebugInfoScreenWrapper = styled(ScreenWrapper)`
+  padding: 0;
+`;
+
+export const MarginedText = styled.Text`
+  margin-top: 8px;
+  margin-bottom: 8px;
```
Comment (kajm) [inline source/screens/debugInfoScreen/DebugInfoScreen.styled.ts:11]:
```js
margin: 8px 0px
```

File: source/screens/debugInfoScreen/DebugInfoScreen.tsx [hunk 14]
```diff
@@ -0,0 +1,57 @@
...
+
+        <DebugInfo debugInfo={debugInfo} />
+
+        <MarginButton
+          block
+          variant="contained"
+          colorSchema="neutral"
+          onClick={async () => {
+            navigation.navigate("App");
+          }}
```
Comment (kajm) [inline source/screens/debugInfoScreen/DebugInfoScreen.tsx:50]: Move this function to a constant instead. And also, the function should not be `async`

```js
const handleBackButtonClick(press = () => {
  navigation.navigate("App");
};
...
      <MarginButton
          block
          variant="contained"
          colorSchema="neutral"
          onClick={handleBackButtonClick}
```

**Review by kajm — CHANGES_REQUESTED** [review]

**Review by kajm — APPROVED** [review]
Borken, but it wörks

**Pull request merged**